services/ux-backend/handlers/rotatekeys/handler.go

```diff
@@ -0,0 +1,89 @@
+package rotatekeys
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	contentTypeTextPlain                     = "text/plain"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+	onboardingValidationPrivateKeySecretName = "onboarding-private-key"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
+		handlePost(r.Context(), w, cl, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(ctx context.Context, w http.ResponseWriter, cl client.Client, namespace string) {
+
+	w.Header().Set("Content-Type", contentTypeTextPlain)
+
+	privateKeySecret := &corev1.Secret{}
+	privateKeySecret.Name = onboardingValidationPrivateKeySecretName
+	privateKeySecret.Namespace = namespace
+	err := cl.Get(ctx, client.ObjectKeyFromObject(privateKeySecret), privateKeySecret)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			klog.Info("onboarding private key secret does not exist")
+			w.WriteHeader(http.StatusBadRequest)
```

Comment:
Why is that a `BadRequest`? Is there anything in the request itself that invalidates it? 
To me, this seems more like a `StatusInternalServerError` situation

Reply:
Based on another comment, this will cease to be a `BadRequest`. My initial intention is this ep shouldn't be triggered for customer managed public keys.

Reply:
But why do we cate if the private key exists or not? or even if the public key exists or not?

Reply:
If public key was customer managed then we should check for private key existence and then only remove the public key.

However, based on your earlier comment and the fact that we also need to generate token for setting native client this endpoint should just delete the public key.

Reply:
I am not sure why we need to delete the public kay and no just trigger the job. What will be the benefit of deleting the public key? It will just requeue a complete reconcile of the storagecluster.

Reply:
I don't think we can trigger job as we aren't using cronjob or it isn't a template.

I understand the intention though and at the same time, what's more alarming, reconciles in quick succession or rotating keys w/o any delays?

Reply:
> I don't think we can trigger job
- awaiting response

Reply:
@leelavg I still think we don't need to relay on a reconcile to trigger rotation. But for the sake of progress, let go with the current approach while having you open a tech debt capturing the requested change and having the link attached to this PR

Reply:
done. https://github.com/red-hat-storage/ocs-operator/issues/2485